findbugs/src/java/edu/umd/cs/findbugs/detect/IdentityHashCodeOnValueBasedClass.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * FindBugs - Find Bugs in Java programs
+ * Copyright (C) 2003-2008 University of Maryland
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+
+package edu.umd.cs.findbugs.detect;
+
+import edu.umd.cs.findbugs.BugInstance;
+import edu.umd.cs.findbugs.BugReporter;
+import edu.umd.cs.findbugs.OpcodeStack.Item;
+import edu.umd.cs.findbugs.ba.vbc.ValueBasedClassIdentifier;
+import edu.umd.cs.findbugs.bcel.OpcodeStackDetector;
+import edu.umd.cs.findbugs.util.ClassName;
+
+public class IdentityHashCodeOnValueBasedClass extends OpcodeStackDetector {
+
+    private final BugReporter bugReporter;
+
+    public IdentityHashCodeOnValueBasedClass(BugReporter bugReporter) {
+        this.bugReporter = bugReporter;
+    }
+
```

Comment:
At first I though `hasInterestingMethod` was a method existing somewhere. I didn't find it, so [I implemented it myself](https://github.com/CodeFX-org/findbugs/commit/23e476959bb2b3b58f1055e2163ea1eb00d8376f#diff-4777956d7399da5348db12b0d1b22a98R61).